### DISCUSSION TITLE
Invalid hook call in getServerSideProps

### DISCUSSION THREAD

**Discussion by kaisiant** [vercel/next.js#17613] [Help] [answered]
```
export const getServerSideProps: GetServerSideProps = async (context) => {
  const { isLoading, error, data } = useQuery('repoData', () =>
    fetch(
      'https://api.github.com/repos/tannerlinsley/react-query'
    ).then((res) => res.json())
  )
  console.log({ data })

  return {
    props: {
      data,
    },
  }
}
```

I went through the discussion but still cannot find the answer that actually solving this problem.

**Comment by rawat07102** [comment, marked as answer]
can not use hooks in getServerSideProps, but you can fetch the data and pass it to useQuery hook in your component as initial data.
ssr docs for react query:  [https://react-query.tanstack.com/docs/guides/ssr](url)

**Reply by kleviss** [reply]
Could you please consider changing the link to https://react-query.tanstack.com/guides/ssr because the above one is not valid anymore? It looks like it has changed.

**Reply by huykon** [reply]
If I don't use react-query so how can I use a custom hook in getServerSideProps? For example [useDarkMode](https://usehooks.com/useDarkMode/)
